tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,257 @@
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import unittest
+from ConfigParser import ConfigParser
+
+import mock
+
+from liota.dcc_comms.mqtt_dcc_comms import MqttDccComms
+from liota.entities.edge_systems.dell5k_edge_system import Dell5KEdgeSystem
+from liota.lib.transports.mqtt import Mqtt, MqttMessagingAttributes
+
+
+class MqttDccCommsTest(unittest.TestCase):
+    """
+    Unit test cases for MQTT DccComms
+    """
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def setUp(self, mock_connect):
+        """
+        Setup all required parameters for MQTT DCC communication tests
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        # ConfigParser to parse ini file
+        self.config = ConfigParser()
+        self.uuid_file = "/etc/liota/conf/uuid.ini"
+
+        # Create Edge System
+        self.edge_system = Dell5KEdgeSystem("TestGateway")
+
+        self.url = "Broker-IP"
+        self.port = "Broker-Port"
+        self.clean_session = True
+        self.publish_message = "test-message"
+
+        # Creating messaging attributes
+        self.mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic")
+
+        # Instantiate client for DCC communication
+        self.client = MqttDccComms(edge_system_name=self.edge_system.name,
+                                   url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def tearDown(self):
+        """
+        Clean up all parameters used in test cases
+        :return:
+        """
+        self.config = None
+        self.uuid_file = None
+        self.edge_system = None
+        self.url = None
+        self.port = None
+        self.clean_session = None
+        self.publish_message = None
+        self.mqtt_msg_attr = None
+        self.client = None
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_no_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation without messaging attributes and no client ID
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port,
+                     clean_session=self.clean_session)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
+
+        edge_system_name = self.config.get('GATEWAY', 'name')
+        local_uuid = self.config.get('GATEWAY', 'local-uuid')
+
+        self.assertEquals(edge_system_name, self.edge_system.name)
+        self.assertIsNotNone(local_uuid)
```

Comment:
This check should be assertEquals.  Generate uuid for the given edge_system name and check whether it is same as the one written to the file